os/drivers/audio/alc5658.c

```diff
@@ -713,6 +713,25 @@ static int alc5658_shutdown(FAR struct audio_lowerhalf_s *dev)
 	return OK;
 }
 
+/****************************************************************************
+ * Name: alc5658_io_err_cb
+ *
+ * Description:
+ *   Callback function for io error
+ *
+ ****************************************************************************/
+static void alc5658_io_err_cb(FAR struct i2s_dev_s *dev, FAR void *arg, int flags)
+{
+	FAR struct alc5658_dev_s *priv = (FAR struct alc5658_dev_s *)arg;
+
+	/* Call upper callback, let it post msg to user q
+	 * apb is set NULL, okay? Rethink
+	*/
+	auddbg("Received error with flags: 0x%x\n", flags);
+	priv->dev.upper(priv->dev.priv, AUDIO_CALLBACK_IOERR, NULL, flags);
```

Comment:
I think the flags must not be passed directly to upper layer. The upper layers cannot make sense of the flags related to I2S. The codec is responsible for converting the flags into some form that the upper layers can understand. 
My suggestion is to add new error codes in audio.h file for overflow and underflow scenario. This function has to check the I2S error and send appropriate error codes to upper layer. 

Also, need to add code in audio.c for AUDIO_CALLBACK_IOERR.

Reply:
I have a same opinion as kishore's

Reply:
After merge this & kishore's code #1157, I will update code regarding this.

Reply:
Right now, the error flags are defined in i2s.h. flags are populated with the values defined in i2s.h. But based on tinyalsa changes & changes in audio layer, this can be modified.